perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/GroupsManagerBlImpl.java

```diff
@@ -4702,16 +4708,20 @@ public Group createGroupUnion(PerunSession sess, Group resultGroup, Group operan
 		// save group relation
 		groupsManagerImpl.saveGroupRelation(sess, resultGroup, operandGroup, parentFlag);
 
+		List<Member> affectedMembers = getGroupMembers(sess, operandGroup);
+
+		Map<Integer, Map<Integer, MemberGroupStatus>> previousStatuses = getPreviousStatuses(sess, operandGroup, affectedMembers);
```

Comment:
For all other methods, you retrieve previous statuses before the changing operation. However, for this one, you do it after the operation `saveGroupRelation`. Is it ok? Have you tested it?

Reply:
I hope so, in this case, if we got previous statuses before `saveGroupRelation`, we wouldn't have necessarily got the previous status of member who was already in the `resultGroup` and we would possibly generate unnecessary audit message. The `saveGroupRelation` does not change statuses of members or members of groups but it makes sure that `resultGroup` is returned in `getResultGroups` of `operandGroup` when retrieving previous statuses. On the other hand during `removeGroupUnion` we want to get previous statuses before removing the link between groups. In `moveGroup` it's a bit more complicated because there we call `createGroupUnion` or `removeGroupUnion` and we need to retrive the previous statuses before that call, otherwise the statuses could have been changed.

Reply:
thanks for the explanation :)